controllers/link.js

```diff
@@ -1,28 +1,39 @@
 const express = require('express');
+
 const router = express.Router();
 const api = require('../api');
 
-router.post('/', function (req, res, next) {
-    // check first if target already exists (preventing db to be wasted)
-    let target = `${(req.headers.origin || process.env.HOST)}/${req.body.target}`;
-    api(req).get("/link/", { qs: { target: target } }).then(result => {
-        let existingLink = result.data[0];
-        if (!existingLink) {
-            api(req).post("/link/", { json: { target: target } }).then(data => {
-                data.newUrl = `${(req.headers.origin || process.env.HOST)}/link/${data._id}`;
-                res.json(data);
-            });
-        } else {
-            existingLink.newUrl = `${(req.headers.origin || process.env.HOST)}/link/${existingLink._id}`;
-            res.json(existingLink);
-        }
-    }).catch(err => next(err));
+router.post('/', (req, res, next) => {
+	// check first if target already exists (preventing db to be wasted)
+	const target = `${(req.headers.origin || process.env.HOST)}/${req.body.target}`;
+	api(req).get('/link/', { qs: { target } }).then((result) => {
+		const existingLink = result.data[0];
```

Comment:
you may access data[0] only if defined, this can fail, in this case the following if makes no sense. you also should not test for objects